src/views/Entities/Zones.vue

```diff
@@ -282,214 +299,236 @@ export default {
+            this.isEmpty = true;
 
-            console.error(error)
+            console.error(error);
           })
           .finally(() => {
             setTimeout(() => {
-              this.isLoading = false
-            }, process.env.VUE_APP_DATA_TIMEOUT)
-          })
-      }
+              this.isLoading = false;
+            }, process.env.VUE_APP_DATA_TIMEOUT);
+          });
+      };
 
-      getZoneStatus()
+      getZoneStatus();
     },
-    getEntity (entity) {
-      this.entityIsLoading = true
-      this.entityIsEmpty = false
+    getEntity(entity) {
+      this.entityIsLoading = true;
+      this.entityIsEmpty = false;
 
-      const selected = ['type', 'name', 'mesh']
+      const selected = ['type', 'name', 'mesh'];
 
       const promise = new Promise((resolve, reject) => {
         if (entity && entity !== null) {
-          resolve(entity)
+          resolve(entity);
         } else {
-          const error = new Error('Entity is either undefined or not present.')
+          const error = new Error('Entity is either undefined or not present.');
 
-          reject(error)
+          reject(error);
         }
-      })
+      });
 
       if (entity && entity !== null) {
         promise
-          .then(response => {
+          .then((response) => {
             if (response) {
               // get the Zone details from the Zone Insights endpoint
-              this.$api.getZoneOverview(response.name)
-                .then((response) => {
-                  this.tabGroupTitle = `Zone: ${response.name}`
-                  this.entityOverviewTitle = `Zone Overview for ${response.name}`
-
-                  // remove `zoneInsight` from the response since we already
-                  // use it in the Zone Insights tab
-                  const cleanRes = () => {
-                    const src = Object.assign({}, response)
-
-                    delete src.zoneInsight
-
-                    return src
-                  }
-
-                  this.entity = getSome(response, selected)
-                  this.rawEntity = stripTimes(response)
-                  this.yamlEntity = cleanRes()
-                })
+              this.$api.getZoneOverview(response.name).then((response) => {
+                this.tabGroupTitle = `Zone: ${response.name}`;
+                this.entityOverviewTitle = `Zone Overview for ${response.name}`;
+
+                let remoteCpVersion = '-';
+                /**
+                 * Iterate through the subscriptions
+                 */
+                if (
+                  response.zoneInsight.subscriptions &&
+                  response.zoneInsight.subscriptions.length
+                ) {
+                  response.zoneInsight.subscriptions.forEach((item) => {
+                    if (item.version && item.version.kumaCp) {
+                      remoteCpVersion = item.version.kumaCp.version;
+                    }
+                  });
+                }
```

Comment:
```suggestion
                const subscriptions = response.zoneInsight.subscriptions || []
                /**
                 * Iterate through the subscriptions
                 */
                 const remoteCpVersion = subscriptions.reduce((acc, { version }) => {
                   if (version && version.kumaCp) {
                     return version.kumaCp.version
                   }
                   
                   return acc
                 }, '-')
```

Reply:
@bartsmykla , I did the version spreading in one place while parsing the `remote-cp` version.